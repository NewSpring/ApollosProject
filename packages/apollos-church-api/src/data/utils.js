import ApollosConfig from '@apollosproject/config';

const { ROCK, ROCK_MAPPINGS } = ApollosConfig;

export const createAssetUrl = (value) =>
  value.path
    ? value.path
    : `${ROCK.IMAGE_URL}/${
        ROCK_MAPPINGS.ASSET_STORAGE_PROVIDERS[`${value.AssetStorageProviderId}`]
      }/${value.Key}`;

export const fakeFunction = () => 'Fake';
