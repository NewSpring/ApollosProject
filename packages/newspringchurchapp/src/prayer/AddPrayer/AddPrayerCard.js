import React, { memo } from 'react';
import { Image } from 'react-native';
import PropTypes from 'prop-types';
import { H5, styled, Button, Card, CardContent } from '@apollosproject/ui-kit';
import AddPrayerHeader from '.';

const StyledCardContent = styled(() => ({
  alignItems: 'center',
}))(CardContent);

const Description = styled(({ theme }) => ({
  color: theme.colors.text.secondary,
  marginBottom: theme.sizing.baseUnit,
}))(H5);

const AddPrayerButton = styled(({ theme }) => ({
  marginTop: theme.sizing.baseUnit * 2,
  width: '100%',
}))(Button);

const ButtonLabel = styled(({ theme }) => ({
  color: theme.colors.text.primary,
}))(H5);

const AddPrayer = memo(({ imgSrc, title, description }) => (
  <Card>
    <StyledCardContent>
      <AddPrayerHeader imgSrc={imgSrc} title={title} />
      <Description>{description}</Description>
      <AddPrayerButton>
        <ButtonLabel>Add Your Prayer</ButtonLabel>
      </AddPrayerButton>
    </StyledCardContent>
  </Card>
));

AddPrayer.propTypes = {
  imgSrc: Image.propTypes,
  title: PropTypes.string,
  description: PropTypes.string,
};

AddPrayer.defaultProps = {
  title: 'Ask for prayer',
  description:
    "There aren't currently any prayers in your community. Send them one now.",
};

AddPrayer.displayName = 'AddPrayer';

export default AddPrayer;
