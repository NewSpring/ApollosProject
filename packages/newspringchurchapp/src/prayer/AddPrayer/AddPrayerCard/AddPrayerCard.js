import React, { memo } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import {
  BodyText,
  styled,
  Button,
  Card,
  CardContent,
  PaddedView,
  Avatar,
  H3,
} from '@apollosproject/ui-kit';

const StyledCardContent = styled({
  alignItems: 'center',
})(CardContent);

const TitleView = styled(({ theme }) => ({
  marginTop: theme.sizing.baseUnit * 0.5,
  marginBottom: theme.sizing.baseUnit * 2,
}))(View);

const AddPrayerCard = memo(({ imgSrc, title, description, ...props }) => (
  <Card>
    <StyledCardContent>
      <Avatar
        source={imgSrc}
        size={'medium'}
        containerStyle={{ marginRight: 0 }}
      />
      <TitleView>
        <H3>{title}</H3>
      </TitleView>
      <BodyText>{description}</BodyText>
    </StyledCardContent>
    <PaddedView>
      <Button
        title={'Add Prayer'}
        onPress={() => props.navigation.navigate('AddPrayerFormConnected')}
      />
    </PaddedView>
  </Card>
));

AddPrayerCard.propTypes = {
  imgSrc: PropTypes.shape({
    uri: PropTypes.string,
  }),
  title: PropTypes.string,
  description: PropTypes.string,
};

AddPrayerCard.defaultProps = {
  title: 'Ask for prayer',
  description:
    "There aren't currently any prayers in your community. Send them one now.",
};

AddPrayerCard.displayName = 'AddPrayerCard';

export default AddPrayerCard;
