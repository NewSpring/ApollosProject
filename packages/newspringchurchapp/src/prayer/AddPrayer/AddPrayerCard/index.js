import React, { memo } from 'react';
import { Image } from 'react-native';
import PropTypes from 'prop-types';
import { H5, styled, Button, Card, CardContent } from '@apollosproject/ui-kit';
import AddPrayerHeader from '../AddPrayerHeader';

const StyledCardContent = styled(() => ({
  alignItems: 'center',
}))(CardContent);

const Description = styled(({ theme }) => ({
  color: theme.colors.text.secondary,
  marginBottom: theme.sizing.baseUnit,
}))(H5);

const AddPrayerButton = styled(({ theme }) => ({
  marginTop: theme.sizing.baseUnit * 2,
  width: '100%',
}))(Button);

const ButtonLabel = styled(({ theme }) => ({
  color: theme.colors.text.primary,
}))(H5);

const AddPrayerCard = memo(({ imgSrc, description }) => (
  <Card>
    <StyledCardContent>
      <AddPrayerHeader imgSrc={imgSrc} />
      <Description>{description}</Description>
      <AddPrayerButton>
        <ButtonLabel>Add Your Prayer</ButtonLabel>
      </AddPrayerButton>
    </StyledCardContent>
  </Card>
));

AddPrayerCard.propTypes = {
  imgSrc: Image.propTypes,
  description: PropTypes.string,
};

AddPrayerCard.defaultProps = {
  description:
    "There aren't currently any prayers in your community. Send them one now.",
};

AddPrayerCard.displayName = 'AddPrayerCard';

export default AddPrayerCard;
