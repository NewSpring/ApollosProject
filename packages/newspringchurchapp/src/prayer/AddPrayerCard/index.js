import React, { memo } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { get } from 'lodash';

import {
  BodyText,
  styled,
  Button,
  Card,
  CardContent,
  PaddedView,
} from '@apollosproject/ui-kit';

import { useQuery } from 'react-apollo';
import GET_USER_PROFILE from 'newspringchurchapp/src/tabs/connect/getUserProfile';
import PrayerHeader from '../PrayerHeader';

const StyledCardContent = styled(({ theme }) => ({
  alignItems: 'center',
  marginTop: theme.sizing.baseUnit,
}))(CardContent);

const StyledBodyText = styled(() => ({
  textAlign: 'center',
}))(BodyText);

const StyledPrayerHeaderView = styled(({ theme }) => ({
  marginBottom: theme.sizing.baseUnit,
}))(View);

const AddPrayerCard = memo(({ title, description, navigation }) => {
  const { data, loading } = useQuery(GET_USER_PROFILE);

  if (loading) return null;

  const avatarSource = get(data, 'currentUser.profile.photo', { uri: null });

  return (
    <Card>
      <StyledCardContent>
        <StyledPrayerHeaderView>
          <PrayerHeader
            avatarSource={avatarSource}
            avatarSize={'medium'}
            name={title}
          />
        </StyledPrayerHeaderView>
        <StyledBodyText>{description}</StyledBodyText>
      </StyledCardContent>
      <PaddedView>
        <Button
          title={'Add Prayer'}
          onPress={() => navigation.navigate('AddPrayerForm')}
        />
      </PaddedView>
    </Card>
  );
});

AddPrayerCard.propTypes = {
  avatarSource: PropTypes.shape({
    uri: PropTypes.string,
  }),
  title: PropTypes.string,
  description: PropTypes.string,
};

AddPrayerCard.defaultProps = {
  title: 'Ask for prayer',
  description:
    "There aren't currently any prayers in your community. Send them one now.",
};

AddPrayerCard.displayName = 'AddPrayerCard';

export default AddPrayerCard;
