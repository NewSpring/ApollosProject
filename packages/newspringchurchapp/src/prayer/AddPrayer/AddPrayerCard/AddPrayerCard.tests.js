import React from 'react';
import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';
import Providers from 'newspringchurchapp/src/Providers';
import GET_USER_PROFILE from 'newspringchurchapp/src/tabs/connect/getUserProfile';

import AddPrayerCard from '.';

const mocks = [
  {
    request: {
      query: GET_USER_PROFILE,
    },
    result: {
      data: {
        currentUser: { profile: { photo: { uri: 'fakeURL' } } },
      },
    },
  },
];

describe('The AddPrayerCard component', () => {
  it('should render', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={mocks}>
        <AddPrayerCard />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom avatar', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={mocks}>
        <AddPrayerCard
          avatarSource={{ uri: 'https://picsum.photos/55/55?random' }}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a custom title', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={mocks}>
        <AddPrayerCard
          description={'this is a custom description for the card.'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
