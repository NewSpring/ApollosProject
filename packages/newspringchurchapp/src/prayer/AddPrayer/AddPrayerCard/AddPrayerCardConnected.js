import React, { memo } from 'react';
import { Query } from 'react-apollo';
import { get } from 'lodash';
import getProfilePic from '../../getProfilePic';
import AddPrayerCard from './AddPrayerCard';

const AddPrayerCardConnected = memo(({ ...props }) => (
  <Query query={getProfilePic} fetchPolicy={'cache-only'}>
    {({ data }) => (
      <AddPrayerCard
        imgSrc={get(data, 'currentUser.profile.photo', { uri: '' })}
        {...props}
      />
    )}
  </Query>
));

AddPrayerCardConnected.displayName = 'AddPrayerCardConnected';

export default AddPrayerCardConnected;
