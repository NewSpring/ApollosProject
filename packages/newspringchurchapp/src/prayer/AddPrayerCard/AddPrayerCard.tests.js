import React from 'react';
import Providers from 'newspringchurchapp/src/Providers';
import GET_USER_PROFILE from 'newspringchurchapp/src/tabs/connect/getUserProfile';
import renderer from 'react-test-renderer';
import AddPrayerCard from '.';

const mocks = [
  {
    request: {
      query: GET_USER_PROFILE,
    },
    result: {
      data: {
        currentUser: { profile: { photo: { uri: 'fakeURL' } } },
      },
    },
  },
];

describe('The AddPrayerCard component', () => {
  it('should render', async () => {
    const tree = renderer.create(
      <Providers mocks={mocks}>
        <AddPrayerCard />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render a custom avatar', async () => {
    const tree = renderer.create(
      <Providers mocks={mocks}>
        <AddPrayerCard
          avatarSource={{ uri: 'https://picsum.photos/55/55?random' }}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render a custom title', async () => {
    const tree = renderer.create(
      <Providers mocks={mocks}>
        <AddPrayerCard
          description={'this is a custom description for the card.'}
        />
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
});
