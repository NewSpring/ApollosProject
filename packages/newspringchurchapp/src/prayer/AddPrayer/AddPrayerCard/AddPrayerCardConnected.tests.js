import React from 'react';

import Providers from 'newspringchurchapp/src/Providers';
import { renderWithApolloData } from 'newspringchurchapp/src/utils/testUtils';
import getUserProfile from 'newspringchurchapp/src/tabs/connect/getUserProfile';
import AddPrayerCardConnected from './AddPrayerCardConnected';

const mocks = [
  {
    request: {
      query: getUserProfile,
    },
    result: {
      data: {
        currentUser: { profile: { photo: { uri: 'fakeURL' } } },
      },
    },
  },
];

it('renders AddPrayerCardConnected', async () => {
  const tree = await renderWithApolloData(
    <Providers mocks={mocks}>
      <AddPrayerCardConnected />
    </Providers>
  );
  expect(tree).toMatchSnapshot();
});
