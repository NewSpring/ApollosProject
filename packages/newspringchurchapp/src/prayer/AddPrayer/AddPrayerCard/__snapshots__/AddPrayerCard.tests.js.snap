// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AddPrayerCard component should render 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "flex-end",
                  "marginBottom": 10,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 40,
                }
              }
              themeSize={40}
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={40}
                bbWidth={40}
                height={40}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    null,
                    Object {
                      "flex": 0,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={40}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                      "fillRule",
                      "stroke",
                      "strokeWidth",
                      "strokeOpacity",
                      "strokeDasharray",
                      "strokeDashoffset",
                      "strokeLinecap",
                      "strokeLinejoin",
                      "strokeMiterlimit",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-3.33 8.33C8.67 8.5 10.2 7 12 7s3.33 1.5 3.33 3.33v.84c0 1.84-1.53 3.33-3.33 3.33S8.67 13 8.67 11.17v-.84zm3.33 10c-2.03 0-3.9-.73-5.33-1.93.7-1.32 2.07-2.23 3.66-2.23h3.34c1.6 0 2.97.9 3.66 2.22-1.44 1.2-3.3 1.92-5.33 1.92z"
                    fill={
                      Array [
                        0,
                        4292730333,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Black",
                  "fontSize": 24,
                  "lineHeight": 27.6,
                  "marginBottom": 32,
                  "marginTop": 8,
                }
              }
            >
              Ask for prayer
            </Text>
          </View>
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
                "marginBottom": 16,
              }
            }
          >
            There aren't currently any prayers in your community. Send them one now.
          </Text>
          <View
            bordered={false}
            disabled={false}
            pill={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6bac43",
                "borderColor": "#6bac43",
                "borderRadius": 48,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Colfax-Medium",
                  "fontSize": 14,
                  "lineHeight": 20.16,
                }
              }
            >
              Add Your Prayer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`The AddPrayerCard component should render a custom avatar 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "flex-end",
                  "marginBottom": 10,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 40,
                }
              }
              themeSize={40}
            >
              <View
                animate={null}
                customAnimate={null}
                onReady={false}
                style={
                  Object {
                    "aspectRatio": 1,
                    "backgroundColor": "#dddddd",
                    "width": "100%",
                  }
                }
              >
                <Image
                  onLoad={[Function]}
                  source={
                    Array [
                      Object {
                        "cache": "force-cache",
                        "uri": "https://picsum.photos/55/55?random",
                      },
                    ]
                  }
                  style={
                    Object {
                      "backgroundColor": "#dddddd",
                      "borderRadius": 20,
                      "bottom": 0,
                      "height": "100%",
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": "100%",
                    }
                  }
                  themeSize={40}
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Black",
                  "fontSize": 24,
                  "lineHeight": 27.6,
                  "marginBottom": 32,
                  "marginTop": 8,
                }
              }
            >
              Ask for prayer
            </Text>
          </View>
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
                "marginBottom": 16,
              }
            }
          >
            There aren't currently any prayers in your community. Send them one now.
          </Text>
          <View
            bordered={false}
            disabled={false}
            pill={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6bac43",
                "borderColor": "#6bac43",
                "borderRadius": 48,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Colfax-Medium",
                  "fontSize": 14,
                  "lineHeight": 20.16,
                }
              }
            >
              Add Your Prayer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`The AddPrayerCard component should render a custom title 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 16,
          "marginHorizontal": 16,
          "marginVertical": 12,
          "shadowColor": "rgba(133, 133, 133, 0.30000000000000004)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 6,
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": undefined,
            "borderRadius": 16,
            "overflow": "hidden",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderRadius": 20,
                  "height": 40,
                  "justifyContent": "flex-end",
                  "marginBottom": 10,
                  "marginRight": 10,
                  "overflow": "hidden",
                  "width": 40,
                }
              }
              themeSize={40}
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={40}
                bbWidth={40}
                height={40}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    null,
                    Object {
                      "flex": 0,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={40}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                      "fillOpacity",
                      "fillRule",
                      "stroke",
                      "strokeWidth",
                      "strokeOpacity",
                      "strokeDasharray",
                      "strokeDashoffset",
                      "strokeLinecap",
                      "strokeLinejoin",
                      "strokeMiterlimit",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm-3.33 8.33C8.67 8.5 10.2 7 12 7s3.33 1.5 3.33 3.33v.84c0 1.84-1.53 3.33-3.33 3.33S8.67 13 8.67 11.17v-.84zm3.33 10c-2.03 0-3.9-.73-5.33-1.93.7-1.32 2.07-2.23 3.66-2.23h3.34c1.6 0 2.97.9 3.66 2.22-1.44 1.2-3.3 1.92-5.33 1.92z"
                    fill={
                      Array [
                        0,
                        4292730333,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "Colfax-Black",
                  "fontSize": 24,
                  "lineHeight": 27.6,
                  "marginBottom": 32,
                  "marginTop": 8,
                }
              }
            >
              Ask for prayer
            </Text>
          </View>
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
                "marginBottom": 16,
              }
            }
          >
            this is a custom description for the card.
          </Text>
          <View
            bordered={false}
            disabled={false}
            pill={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#6bac43",
                "borderColor": "#6bac43",
                "borderRadius": 48,
                "borderWidth": 2,
                "elevation": 2,
                "flexDirection": "row",
                "height": 48,
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
                "overflow": "hidden",
                "paddingHorizontal": 16,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Colfax-Medium",
                  "fontSize": 14,
                  "lineHeight": 20.16,
                }
              }
            >
              Add Your Prayer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
